aten/src/ATen/core/LegacyTypeDispatch.h

```diff
@@ -84,6 +84,34 @@ struct TORCH_API AutoNonVariableTypeMode {
   c10::impl::ExcludeDispatchKeyGuard autograd_guard_;
 };
 
+struct TORCH_API AutoDispatchBelowFunctionalize {
+  AutoDispatchBelowFunctionalize() :
+    dispatch_key_guard_(c10::DispatchKeySet(c10::DispatchKey::Functionalize)) {
```

Comment:
Does it matter that this masks out only `Functionalize`, rather than it and everything above?

Reply:
I can probably just delete this and directly use `ExcludeDispatchKeyGuard(DispatchKey::Functionalize)`

I don't think we want it to mask out stuff above, since there could be random other dispatcher functionality that expects to be re-runnable when you call into other tensor ops. All the functionalize kernels should care about is making sure that they don't re-enter other functionalize kernels.